go.mod

```diff
@@ -46,56 +50,186 @@ require (
 )
 
 require (
+	cloud.google.com/go v0.115.0 // indirect
+	cloud.google.com/go/auth v0.5.1 // indirect
+	cloud.google.com/go/auth/oauth2adapt v0.2.2 // indirect
+	cloud.google.com/go/iam v1.1.8 // indirect
+	cloud.google.com/go/kms v1.18.0 // indirect
+	cloud.google.com/go/longrunning v0.5.7 // indirect
+	contrib.go.opencensus.io/exporter/ocagent v0.7.1-0.20200907061046-05415f1de66d // indirect
+	contrib.go.opencensus.io/exporter/prometheus v0.4.2 // indirect
+	cuelang.org/go v0.8.2 // indirect
+	github.com/AliyunContainerService/ack-ram-tool/pkg/credentials/alibabacloudsdkgo/helper v0.2.0 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.12.0 // indirect
```

Comment:
Just noticing these additional dependencies, which are unfortunate :( 

```
❯ go mod why -m github.com/Azure/azure-sdk-for-go/sdk/azcore
# github.com/Azure/azure-sdk-for-go/sdk/azcore
github.com/crossplane/crossplane/internal/controller/pkg/signature
github.com/sigstore/policy-controller/pkg/webhook
github.com/sigstore/policy-controller/pkg/webhook/registryauth
github.com/sigstore/policy-controller/pkg/webhook/registryauth/azure
github.com/Azure/azure-sdk-for-go/sdk/azcore/policy
```

Reply:
Still a bunch of new dependencies, but I think I did all I could do here, see https://github.com/crossplane/crossplane/pull/6022/commits/dc6ebc9674e044610b53c6c04a229d27167bc158#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

I basically replaced the policy controller's [ValidatePolicy](https://github.com/sigstore/policy-controller/blob/6f637fe6cbaf09490d1c9e5b068609463d1bf9e3/pkg/webhook/validator.go#L477) function call with my [own implementation](https://github.com/crossplane/crossplane/pull/6022/files#diff-d019a84f3bd4bc2bfd20016691b1bf0d61f1b4544c5cbc0a529c6c1460c2130fR34), directly calling lower-level methods, and avoided importing packages whenever I could [implement on my own](https://github.com/crossplane/crossplane/pull/6022/files#diff-d019a84f3bd4bc2bfd20016691b1bf0d61f1b4544c5cbc0a529c6c1460c2130fR184) or [bring in the function](https://github.com/crossplane/crossplane/pull/6022/files#diff-bb78683fbdec4d8ff2e1b2220f312298d3c02bb7bfbc032aa530afdd605b80deR16) while keeping its license.

TBH, apart from reducing the number of additional dependencies since it ended up with a simpler implementation not dependending on policy controller's internal types. We lost some functionality, like using custom [fulcio](https://docs.sigstore.dev/certificate_authority/overview/) instances etc., but I believe we have enough functionality for a first iteration. See e2e's for what is covered.